### PR TITLE
Expose additional methods from HttpWebRequest and Response

### LINKED ISSUE #13900: HttpWebRequest and HttpWebResonse - expose missing members
Implement + expose + test these methods:
```
P:System.Net.HttpWebRequest.Connection
P:System.Net.HttpWebRequest.ContentLength
P:System.Net.HttpWebRequest.DefaultCachePolicy
P:System.Net.HttpWebRequest.PreAuthenticate
P:System.Net.HttpWebRequest.Proxy
P:System.Net.HttpWebRequest.Timeout
M:System.Net.HttpWebRequest.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
M:System.Net.HttpWebRequest.GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
M:System.Net.HttpWebRequest.System#Runtime#Serialization#ISerializable#GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)

P:System.Net.HttpWebResponse.IsMutuallyAuthenticated
M:System.Net.HttpWebResponse.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
M:System.Net.HttpWebResponse.GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
M:System.Net.HttpWebResponse.System#Runtime#Serialization#ISerializable#GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
```

### PR DESCRIPTION
**Pull request by CIPop**
Fixes #13900 

This PR contains both contributions made by @tijoytom in  #13441 (already reviewed in that PR). 
I'm addressing the PR comments and creating a new one while he's OOF.

@stephentoub @davidsh @karelz PTAL

### CODE REVIEW

File: src/System.Net.Requests/src/System/Net/HttpWebRequest.cs [hunk 1]
```diff
@@ -1131,12 +1184,21 @@ private Task<Stream> GetRequestStreamTask()
...
                 handler.MaxResponseHeadersLength = MaximumResponseHeadersLength;
+                handler.PreAuthenticate = PreAuthenticate;
+                if (Timeout == Threading.Timeout.Infinite)
+                {
+                    client.Timeout = Threading.Timeout.InfiniteTimeSpan;
+                }
+                else
+                {
+                    client.Timeout = TimeSpan.FromMilliseconds(Timeout);                    
+                }
```
Comment (stephentoub) [inline src/System.Net.Requests/src/System/Net/HttpWebRequest.cs]: Nit: could be condensed to:
```C#
client.Timeout = Timeout == Threading.Timeout.Infinite ?
    Threading.Timeout.InfiniteTimeSpan :
    TimeSpan.FromMilliseconds(Timeout);
```

File: src/System.Net.Requests/tests/HttpWebRequestHeaderTest.cs [hunk 2]
```diff
@@ -25,12 +29,12 @@ public void Ctor_VerifyHttpRequestDefaults(Uri remoteServer)
...
             Assert.Null(request.CookieContainer);
-            Assert.Equal(0,HttpWebRequest.DefaultMaximumErrorResponseLength);
-            Assert.Equal(65536,HttpWebRequest.DefaultMaximumResponseHeadersLength);
+            Assert.Equal(0, HttpWebRequest.DefaultMaximumErrorResponseLength); // NetFX behavior difference (64 on NetFX).
+            Assert.Equal(65536, HttpWebRequest.DefaultMaximumResponseHeadersLength); // NetFX behavior difference (64 on NetFX).
             Assert.Null(request.CookieContainer);
             Assert.True(request.AllowWriteStreamBuffering);
             Assert.NotNull(request.ClientCertificates);
-            Assert.Throws<NotImplementedException>(() => request.ConnectionGroupName);
+            Assert.Throws<NotImplementedException>(() => request.ConnectionGroupName); //NetFX behavior difference.
```
Comment (stephentoub) [inline src/System.Net.Requests/tests/HttpWebRequestHeaderTest.cs]: Should these behavioral differences be separated out into a separate test attributed with SkipOnTarget for NetFramework?

File: src/System.Net.Requests/tests/HttpWebRequestHeaderTest.cs [hunk 3]
```diff
@@ -97,9 +95,110 @@ public void HttpWebRequest_EndGetRequestStreamContext_Null(Uri remoteServer)
...
+                Uri uri = new Uri($"http://{ep.Address}:{ep.Port}");
+
+                HttpWebRequest request = WebRequest.CreateHttp(uri);
+                request.Timeout = 10; // ms.
+                
+                var sw = Stopwatch.StartNew();
+                WebException exception = Assert.Throws<WebException>(() =>
+                {
+                    var response = (HttpWebResponse)request.GetResponse();
+                });
```
Comment (stephentoub) [inline src/System.Net.Requests/tests/HttpWebRequestHeaderTest.cs]: Nit: we don't need to cast / store the response, as we expect it to throw.

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
@dotnet-bot test this please

File: src/System.Net.Requests/src/System/Net/HttpWebRequest.cs [hunk 4]
```diff
@@ -272,7 +274,46 @@ public int ContinueTimeout
...
+
+                _timeout = value;
+            }
+        }
+
+        public override long ContentLength
+        {
+            get
+            {
+                long value = 0;
```
Comment (CIPop) [inline src/System.Net.Requests/src/System/Net/HttpWebRequest.cs]: Thanks! Fixed.

File: src/System.Net.Requests/src/System/Net/HttpWebRequest.cs (hunk 1 above)
Comment (CIPop) [inline src/System.Net.Requests/src/System/Net/HttpWebRequest.cs]: Fixed.

File: src/System.Net.Requests/tests/HttpWebRequestHeaderTest.cs (hunk 2 above)
Comment (CIPop) [inline src/System.Net.Requests/tests/HttpWebRequestHeaderTest.cs]: We need to first enable the tests to run on NetFramework. 
Tracking with #14101.

**Comment by CIPop** [comment]
@dotnet-bot test this please

**Comment by CIPop** [comment]
@mmitche /cc @karelz FYI: the jobs were in the "Triggered" mode for a while (hours) but not starting. Not sure if this is known.
I see now that more started but CentOS is still "Triggered." after a minute or so.

**Comment by karelz** [comment]
I think it happens sometimes and that @mmitche is aware of it ...
cc @stephentoub

**Comment by danmosemsft** [comment]
@CIPop looks like yo'ure good to go

**Pull request merged**